src/test/regress/sql/multi_real_time_transaction.sql

```diff
@@ -220,14 +224,17 @@ SET LOCAL citus.multi_shard_modify_mode TO 'sequential';
 SELECT id, pg_advisory_lock(15) FROM test_table ORDER BY 1 DESC;
 ROLLBACK;
 
+
 SET client_min_messages TO DEFAULT;
 alter system set deadlock_timeout TO DEFAULT;
 SELECT pg_reload_conf();
 
 BEGIN;
 SET citus.select_opens_transaction_block TO off;
 -- This query would self-deadlock if it ran in a distributed transaction
-SELECT id, pg_advisory_lock(15) FROM test_table ORDER BY id;
+-- we use a different advisory lock because previous tests
```

Comment:
A sad story that took me really long to understand what is going on. Basically, since the executor does not disconnect (e.g., due to `SESSION_LIFESPAN`), the lock acquired by the above test leads to blocking this one.

So, an example of what I hit where `ROLLBACK` doesn't release `pg_advisory_lock `. Well, that's obvious from the documentation, but still it was very confusing for me because previously real-time executor always disconnects so we haven't had this issue at all. I mostly thought that this is a bug in the new executor, which turned out to be wrong...



```SQL
BEGIN
SELECT  pg_advisory_lock(15);
 pg_advisory_lock 
------------------
 
(1 row)

SELECT locktype, objid, mode, granted FROM pg_locks where pid = pg_backend_pid();
  locktype  | objid |      mode       | granted 
------------+-------+-----------------+---------
 relation   |       | AccessShareLock | t
 relation   |       | AccessShareLock | t
 virtualxid |       | ExclusiveLock   | t
 advisory   |    15 | ExclusiveLock   | t
(4 rows)

ROLLBACK;

SELECT locktype, objid, mode, granted FROM pg_locks where pid = pg_backend_pid();
  locktype  | objid |      mode       | granted 
------------+-------+-----------------+---------
 relation   |       | AccessShareLock | t
 virtualxid |       | ExclusiveLock   | t
 advisory   |    15 | ExclusiveLock   | t
(3 rows)
```